designs/aws-launch-template-network-interfaces.md

```diff
@@ -0,0 +1,126 @@
+      interfaceType: "interface"          # optional, The interface type for the network interface. To create an Elastic Fabric Adapter (EFA), specify efa
+      networkCardIndex: 0 
+      ipv4PrefixCount: 1                  # optional, The number of IPv4 prefixes that AWS automatically assigns to the network interface.
+      ipv6PrefixCount: 1                  # optional, The number of IPv6 prefixes that AWS automatically assigns to the network interface.
+    - { ... }
+```
+
+subnet and securityGroup will remain part of the `AWSNodeTemplate` and will be used for both network interfaces.
+in following iterations, as the `AWSNodeTemplate` move from `v1alpha1` to higher versions, we can add support for:
+
+```yaml
+subnetSelector: String
+securityGroupSelector: String
+```
+
+this way we can support different subnets and security groups for each network interface, we can also remove the limit
+of having only 2 network interfaces per LaunchTemplate.
+
+The [drift design doc #366](https://github.com/aws/karpenter-core/pull/366) suggests classifying CRD fields as  (1) Static, (2) Dynamic, or (3) Behavioral. in this
+context, the networkInterfaces would be classified as following:
+
+|                                                                | Static | Dynamic | Behavioral |
+|----------------------------------------------------------------|--------|---------|------------|
+| AWSNodeTemplate.NetworkInterfaces.length                       | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces.order                        | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].AssociateCarrierIpAddress | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].AssociatePublicIpAddress  | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].DeleteOnTermination       |        |         | x          |
+| AWSNodeTemplate.NetworkInterfaces[@].Description               | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].DeviceIndex               | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].InterfaceType             | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].NetworkCardIndex          | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].ipv4PrefixCount           | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].ipv6PrefixCount           | x      |         |            |
+
+since `AWSNodeTemplate.NetworkInterfaces` is a list, Drift would be triggered if:
+
+- the number of network interfaces of the node does not contain the specified `AWSNodeTemplate.NetworkInterfaces`
+- the order of network interfaces of a node do not match the order of `AWSNodeTemplate.NetworkInterfaces` when sorted
+  by `NetworkCardIndex` and `DeviceIndex`.
+  `AWSNodeTemplate.NetworkInterfaces` do not define `NetworkCardIndex` or `DeviceIndex`, their values are assumed to
```

Comment:
Sorry if I'm missing the point here, but I see that you've included both NetworkCardIndex and DeviceIndex in the specifications. Do you mean to add an if here? 
```suggestion
  If `AWSNodeTemplate.NetworkInterfaces` do not define `NetworkCardIndex` or `DeviceIndex`, their values are assumed to
```

Reply:
yes, thanks for spotting it